static/app/stories/view/storySearch.tsx

```diff
@@ -1,100 +1,138 @@
-        },
-        {replace: true}
-      );
-    },
-    [location.query, navigate]
-  );
-
+  const tree = useStoryTree(files, {query: '', representation: 'category', type: 'flat'});
   const storiesSearchHotkeys = useMemo(() => {
-    return [{match: '/', callback: () => searchInput.current?.focus()}];
+    return [{match: '/', callback: () => inputRef.current?.focus()}];
   }, []);
   useHotkeys(storiesSearchHotkeys);
+  const location = useLocation();
+  const defaultSelectedKey = location.query.name as string;
+
+  return (
+    <SearchComboBox
+      label={t('Search stories')}
+      defaultItems={tree}
+      defaultSelectedKey={defaultSelectedKey}
+      menuTrigger="focus"
+      listBoxRef={listBoxRef}
+      labelRef={labelRef}
+      inputRef={inputRef}
+      popoverRef={popoverRef}
+      items={tree}
+    >
+      {tree}
+    </SearchComboBox>
+  );
+}
+
+function SearchInput(
+  props: React.HTMLProps<HTMLInputElement> & React.RefAttributes<HTMLInputElement>
+) {
+  const {className: _0, style: _1, size: nativeSize, ...nativeProps} = props;
+
+  return (
+    <InputGroup style={{minHeight: 33, height: 33, width: 256}}>
+      <InputGroup.LeadingItems disablePointerEvents>
+        <IconSearch />
+      </InputGroup.LeadingItems>
+      <InputGroup.Input ref={props.ref} nativeSize={nativeSize} {...nativeProps} />
+      <InputGroup.TrailingItems>
+        <Badge type="internal">/</Badge>
+      </InputGroup.TrailingItems>
+    </InputGroup>
+  );
+}
+
+interface SearchComboBoxProps<T extends StoryTreeNode>
+  extends Omit<AriaComboBoxProps<T>, 'children'> {
+  children: Iterable<T>;
+  inputRef: React.RefObject<HTMLInputElement | null>;
+  labelRef: React.RefObject<HTMLElement | null>;
+  listBoxRef: React.RefObject<HTMLElement | null>;
+  popoverRef: React.RefObject<HTMLElement | null>;
+  description?: string | null;
+  label?: string;
+}
+
+function SearchComboBox<T extends StoryTreeNode>({
+  label,
+  inputRef,
+  listBoxRef,
+  popoverRef,
+  children,
+  defaultItems,
+  ...props
+}: SearchComboBoxProps<T>) {
+  // const navigate = useNavigate();
+  const state = useComboBoxState(props);
+  const listState = useListState<StoryTreeNode>({
+    items: defaultItems,
+    selectionMode: 'single',
+    selectionBehavior: 'replace',
+    disallowEmptySelection: true,
+    *filter(nodes) {
+      if (!state.inputValue) {
+        yield* nodes;
+        return;
+      }
+      for (const node of nodes) {
+        if (node.textValue.includes(state.inputValue)) {
+          yield node;
+        }
+      }
+    },
+  });
+  const {labelProps, inputProps, listBoxProps} = useComboBox(
+    {...props, items: children, inputRef, popoverRef, listBoxRef},
+    state
+  );
+  // const handleAction = useCallback(
+  //   (key: string | number | boolean | null) => {
+  //     if (!key) {
+  //       return;
+  //     }
+  //     navigate(`/stories?name=${encodeURIComponent(key)}`, {replace: true});
+  //   },
+  //   [navigate]
+  // );
```

Comment:
Is this still needed?

```suggestion
  // const handleAction = useCallback(
  //   (key: string | number | boolean | null) => {
  //     if (!key) {
  //       return;
  //     }
  //     navigate(`/stories?name=${encodeURIComponent(key)}`, {replace: true});
  //   },
  //   [navigate]
  // );
```

Reply:
Fixed in a4ac454690d